src/perf/bin/drvmain.cpp

```diff
@@ -0,0 +1,604 @@
+    }
+}
+
+void* __cdecl operator new[](size_t Size) {
+    return ExAllocatePool2(POOL_FLAG_NON_PAGED, Size, QUIC_PERF_TAG);
+}
+
+void __cdecl operator delete[](_In_opt_ void* Mem) {
+    if (Mem != nullptr) {
+        ExFreePoolWithTag(Mem, QUIC_PERF_TAG);
+    }
+}
+
+extern "C" _IRQL_requires_max_(PASSIVE_LEVEL) void QuicTraceRundown(void) { }
+
+extern "C"
+INITCODE
+_Function_class_(DRIVER_INITIALIZE)
+_IRQL_requires_same_
+_IRQL_requires_(PASSIVE_LEVEL)
+NTSTATUS
+DriverEntry(
+    _In_ PDRIVER_OBJECT DriverObject,
+    _In_ PUNICODE_STRING RegistryPath
+)
+/*++
+
+Routine Description:
+
+    DriverEntry initializes the driver and is the first routine called by the
+    system after the driver is loaded. DriverEntry specifies the other entry
+    points in the function driver, such as EvtDevice and DriverUnload.
+
+Parameters Description:
+
+    DriverObject - represents the instance of the function driver that is loaded
+    into memory. DriverEntry must initialize members of DriverObject before it
+    returns to the caller. DriverObject is allocated by the system before the
+    driver is loaded, and it is released by the system after the system unloads
+    the function driver from memory.
+
+    RegistryPath - represents the driver specific path in the Registry.
+    The function driver can use the path to store driver related data between
+    reboots. The path does not store hardware instance specific data.
+
+Return Value:
+
+    A success status as determined by NT_SUCCESS macro, if successful.
+
+--*/
+{
+    NTSTATUS Status;
+    WDF_DRIVER_CONFIG Config;
+    WDFDRIVER Driver;
+    BOOLEAN PlatformInitialized = FALSE;
+
+    QuicPlatformSystemLoad(DriverObject, RegistryPath);
+
+    Status = QuicPlatformInitialize();
+    if (!NT_SUCCESS(Status)) {
+        QuicTraceEvent(
+            LibraryErrorStatus,
+            "[ lib] ERROR, %u, %s.",
+            Status,
+            "QuicPlatformInitialize failed");
+        goto Error;
+    }
+    PlatformInitialized = TRUE;
+
+    //
+    // Create the WdfDriver Object
+    //
+    WDF_DRIVER_CONFIG_INIT(&Config, NULL);
+    Config.EvtDriverUnload = QuicPerfDriverUnload;
+    Config.DriverInitFlags = WdfDriverInitNonPnpDriver;
+    Config.DriverPoolTag = QUIC_PERF_TAG;
+
+    Status =
+        WdfDriverCreate(
+            DriverObject,
+            RegistryPath,
+            WDF_NO_OBJECT_ATTRIBUTES,
+            &Config,
+            &Driver);
+    if (!NT_SUCCESS(Status)) {
+        QuicTraceEvent(
+            LibraryErrorStatus,
+            "[ lib] ERROR, %u, %s.",
+            Status,
+            "WdfDriverCreate failed");
+        goto Error;
+    }
+
+    //
+    // Initialize the device control interface.
+    //
+    Status = QuicPerfCtlInitialize(Driver);
+    if (!NT_SUCCESS(Status)) {
+        QuicTraceEvent(
+            LibraryErrorStatus,
+            "[ lib] ERROR, %u, %s.",
+            Status,
+            "QuicPerfCtlInitialize failed");
```

Comment:
Unnecessary, as the function will log at the actual point of failure. In general, we only log function failure if it's an external call.

Reply:
Copied straight from the BVT code. We should create an issue to fix the issues seen here over there too.